tfe/tool_helpers.go

```diff
@@ -19,5 +20,37 @@ func fetchTerraformVersionID(version string, client *tfe.Client) (string, error)
 		return "", fmt.Errorf("terraform version not found")
 	}
 
-	return versions.Items[0].ID, nil
+	if versions.Items[0].Version != version {
+
+		options := tfe.AdminTerraformVersionsListOptions{}
+
+	found:
+		for {
+			vlist, err := client.Admin.TerraformVersions.List(ctx, options)
+			if err != nil {
+				return "", fmt.Errorf("error reading Terraform Versions: %w", err)
+			}
+
+			if vlist.CurrentPage >= vlist.TotalPages {
+				break
+			}
```

Comment:
I think you've put this too early! It should go after the inner search loop, otherwise it's impossible to ever find a version that happens to be on the final page of the list. Cf. prior art in readWorkspaceStateConsumers, for example.